pkg/operation/botanist/controlplane.go

```diff
@@ -670,20 +671,50 @@ func (b *Botanist) DeployNetworkPolicies(ctx context.Context) error {
 // DeployKubeAPIServerService deploys kube-apiserver service.
 func (b *Botanist) DeployKubeAPIServerService(ctx context.Context) error {
 	const name = "kube-apiserver-service"
-	return b.ChartApplierSeed.Apply(ctx, filepath.Join(chartPathControlPlane, name), b.Shoot.SeedNamespace, name)
+
+	values := map[string]interface{}{
+		"kubernetesVersion": b.Shoot.Info.Spec.Kubernetes.Version,
+	}
+	return b.ChartApplierSeed.Apply(ctx, filepath.Join(chartPathControlPlane, name), b.Shoot.SeedNamespace, name, kubernetes.Values(values))
 }
 
 // DeployKubeAPIServer deploys kube-apiserver deployment.
 func (b *Botanist) DeployKubeAPIServer(ctx context.Context) error {
-	hvpaEnabled := gardenletfeatures.FeatureGate.Enabled(features.HVPA)
-	memoryMetricForHpaEnabled := false
+	var (
+		hvpaEnabled               = gardenletfeatures.FeatureGate.Enabled(features.HVPA)
+		memoryMetricForHpaEnabled = false
+	)
+
+	kubernetesVersionGeq118, err := versionutils.CheckVersionMeetsConstraint(b.Shoot.KubernetesMajorMinorVersion, ">= 1.18")
```

Comment:
Probably it makes sense to introduce a field in the `pkg/operation/shoot` that tells you whether to use `Konnectivity` or not. Then you can compute it only once at the beginning and use it everywhere (yes, it's not too expensive to compute it multiple times, though, it'd be easier to read).